File: react-native-nitro-player/ios/core/NitroPlayerLogger.swift
//
//  Logger.swift
//  NitroPlayer
//
//  Created by Ritesh Shukla on 18/02/26.
//

import Foundation

class NitroPlayerLogger {
    #if DEBUG
    static var isEnabled = true
    #else
    static var isEnabled = false
    #endif

    static func log(_ header: String = "NitroPlayer", _ message: @autoclosure () -> String) {
        if isEnabled {
            print("[\(header)] \(message())")
        }
    }
}

File: react-native-nitro-player/android/src/main/java/com/margelo/nitro/nitroplayer/core/NitroPlayerLogger.kt
package com.margelo.nitro.nitroplayer.core

import android.util.Log
import com.margelo.nitro.nitroplayer.BuildConfig

object NitroPlayerLogger {
    // Determine if logging is enabled based on build config
    val isEnabled: Boolean = BuildConfig.DEBUG

    fun log(
        header: String = "NitroPlayer",
        message: String,
    ) {
        if (isEnabled) {
            Log.d(header, message)
        }
    }
}
